src/plugins/data_source_management/public/components/data_source_aggregated_view/data_source_aggregated_view.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiNotificationBadge,
+  EuiPopover,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { SavedObjectsClientContract, ToastsStart } from 'opensearch-dashboards/public';
+import { DataSourceOption } from '../data_source_selector/data_source_selector';
+import { getDataSourcesWithFields } from '../utils';
+
+interface DataSourceAggregatedViewProps {
+  savedObjectsClient: SavedObjectsClientContract;
+  notifications: ToastsStart;
+  hideLocalCluster: boolean;
+  fullWidth: boolean;
+  activeDatasourceIds?: string[];
+  filterFn?: (dataSource: any) => boolean;
+  displayAllCompatibleDataSources: boolean;
+}
+
+interface DataSourceAggregatedViewState {
+  dataSourceOptions: DataSourceOption[];
+  selectedOption: DataSourceOption[];
+  isPopoverOpen: boolean;
+  allDataSourcesIdMap: Map<string, any>;
+}
+
+export class DataSourceAggregatedView extends React.Component<
+  DataSourceAggregatedViewProps,
+  DataSourceAggregatedViewState
+> {
+  private _isMounted: boolean = false;
+
+  constructor(props: DataSourceAggregatedViewProps) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false,
+      allDataSourcesIdMap: new Map(),
+    };
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  onClick() {
+    this.setState({ ...this.state, isPopoverOpen: !this.state.isPopoverOpen });
+  }
+
+  closePopover() {
+    this.setState({ ...this.state, isPopoverOpen: false });
+  }
+
+  async componentDidMount() {
+    this._isMounted = true;
+    getDataSourcesWithFields(this.props.savedObjectsClient, ['id', 'title', 'auth.type'])
+      .then((fetchedDataSources) => {
+        if (fetchedDataSources?.length) {
+          let filteredDataSources = fetchedDataSources;
+          if (this.props.filterFn) {
+            filteredDataSources = fetchedDataSources.filter((ds) => this.props.filterFn!(ds));
+          }
+
+          const allDataSourcesIdMap = new Map();
+
+          filteredDataSources.forEach((ds) => {
+            allDataSourcesIdMap.set(ds.id, ds.attributes!.title || '');
+          });
+
+          if (!this.props.hideLocalCluster) {
+            allDataSourcesIdMap.set('', 'Local cluster');
```

Comment:
Q: what's a "local cluster"?

Reply:
@xinruiba might be at better position about this local cluster, basically, it refers to the open search cluster running at local host:9200, and for multiple data source, since it enables us to connect to remote data source, thus we need to be able to distinguish local vs remote

Reply:
That makes sense, thanks for the clarification!